recipes/nanorange/all/conanfile.py

```diff
@@ -11,50 +14,52 @@ class NanorangeConan(ConanFile):
-        version = Version( self.settings.compiler.version )
-        compiler = self.settings.compiler
-        if self.settings.compiler.cppstd and \
-           not any([str(self.settings.compiler.cppstd) == std for std in ["17", "20", "gnu17", "gnu20"]]):
-            raise ConanInvalidConfiguration("nanoRange requires at least c++17")
-        elif compiler == "Visual Studio":
-            if version < "16":
-                raise ConanInvalidConfiguration("NanoRange requires at least Visual Studio version 15.9, please use 16")
-            if not any([self.settings.compiler.cppstd == std for std in ["17", "20"]]):
-                raise ConanInvalidConfiguration("nanoRange requires at least c++17")
-        else:
-            if ( compiler == "gcc" and version < "7" ) or ( compiler == "clang" and version < "5" ):
-                raise ConanInvalidConfiguration("NanoRange requires a compiler that supports at least C++17")
-            elif compiler == "apple-clang":
-                self.output.warn("NanoRange is not tested with apple-clang")
-                if version < "10":
-                    raise ConanInvalidConfiguration("NanoRange requires a compiler that supports at least C++17")
+    def _min_cppstd(self):
+        return 17
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        url = self.conan_data["sources"][self.version]["url"]
-        commit = url[url.rfind("/")+1:url.find(".tar.gz")]
-        extracted_folder = "NanoRange-" + commit
-        os.rename(extracted_folder, self._source_subfolder)
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
 
-    def package(self):
-        self.copy("*.hpp", src=os.path.join(self._source_subfolder, "include"), dst="include")
-        self.copy("LICENSE_1_0.txt", src=self._source_subfolder, dst="licenses")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def package(self):
+        copy(self, pattern="LICENSE_1_0.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(
+            self,
+            pattern="*.h",
```

Comment:
(Forgot to send message in the old review). NanoRange uses `.hpp`. Does this also copy those?

Reply:
Nop! Good catch 👏  just cipy paste error by me :)

```suggestion
            pattern="*.hpp",
```